stockpile/roles/openshift-cluster-topology/tasks/main.yml

```diff
@@ -0,0 +1,66 @@
+---
+- name: check if oc client is installed
+  shell: which oc
+  register: oc_installed
+
+- name: check if kubeconfig exists
+  stat:
+    path: "{{ ansible_env.HOME }}/.kube/config"
+  register: kubeconfig
+
+- name: fail if oc client or kubeconfig doesn't exist
+  fail:
+    msg: Cannot find kubeconfig at "{{ ansible_env.HOME }}/.kube/config", please check.
+  when: ( oc_installed.rc != 0 or kubeconfig.stat.exists == False )
+
+- name: check if jq is installed
+  shell: which jq
+  register: jq_installed
+
+- name: install jq if not installed
+  package:
+    name: jq
+    state: latest
+  when: jq_installed.rc != 0
+
+- block:
+    - name: get number of pods in running state on the cluster
+      shell: oc get pods --all-namespaces -o json | jq '.items[].status.phase' | grep "Running" | wc -l
+      register: ocp_running_pods
+
+    - name: get openshift app node count
+      shell:  oc get nodes -l node-role.kubernetes.io/compute=true | grep -w "Ready" | wc -l
+      register: ocp_compute_count
+
+    - name: openshift masters and etcd count assuming that masters are etcd are co-located
+      shell: oc get nodes -l node-role.kubernetes.io/master=true | grep -w "Ready" | wc -l
+      register: ocp_master_count
+
+    - name: ocp cluster network
+      shell: oc get clusternetwork -o json | jq '.items[].pluginName'
+      register: ocp_network_plugin
+    
+    - name: oc client version
+      shell: oc version | grep -w oc | cut -d ' ' -f2
+      register: oc_client_version
+
+    - name: oc server version
+      shell: oc version | grep -w openshift | cut -d ' ' -f2
+      register: oc_server_version
+
+    - name: openshift user
+      shell: oc whoami
+      register: ocp_user
+
+    - name: set the collected info as facts
+      set_fact:
+        openshift_cluster_topology:
+          - running_pod_count: "{{ ocp_running_pods.stdout }}"
```

Comment:
We don't need to create a dict of lists. We could just create a dict. 

```yaml
openshift_cluster_topology:
    running_pod_count: "{{ ocp_running_pods.stdout }}"
```
^ For example.

Reply:
Makes sense, fixed it.